### PR TITLE
[WIP] Add a SSL recipe on LibriSpeech dataset 

### PR DESCRIPTION
**Pull request by yfyeung**
- [x] Gradient accumulation mechanism
- [x] HuBERT fine-tune recipe based on transformers, which supports the pre-trained checkpoints from:
        - huggingface/facebook: [hubert-base-ls960](https://huggingface.co/facebook/hubert-base-ls960), [hubert-large-ll60k](https://huggingface.co/facebook/hubert-large-ll60k), [hubert-xlarge-ll60k](https://huggingface.co/facebook/hubert-xlarge-ll60k) 
        - huggingface/TencentGameMate: [chinese-hubert-base](https://huggingface.co/TencentGameMate/chinese-hubert-base), [chinese-hubert-large](https://huggingface.co/TencentGameMate/chinese-hubert-large)
- [x] HuBERT pre-train recipe based on transformers
- [ ] Release with the paper

### CODE REVIEW

**Comment by yfyeung** [comment]
The zipformer pre-train recipe will be released along with the well-tuned results.

**Comment by yfyeung** [comment]
The HuBERT finetuning recipe with the TencentGameMate [chinese-hubert-large](https://huggingface.co/TencentGameMate/chinese-hubert-large) was utilized in track 1 of the ICASSP2024 ICMC-ASR Grand Challenge, achieving a CER of 18.18 directly in the evaluation.

**Comment by yfyeung** [comment]
The zipformer pretraining / finetuning recipe will be released along with the paper.

**Comment by v-tuenv** [comment]
@yfyeung Could you provide me with the datasets from the ICASSP 2024 ICMC-ASR Grand Challenge? Unfortunately, the challenge has already concluded, and I am unable to register to experiment with it.

**Comment by yfyeung** [comment]
> @yfyeung Could you provide me with the datasets from the ICASSP 2024 ICMC-ASR Grand Challenge? Unfortunately, the challenge has already concluded, and I am unable to register to experiment with it.

Sure
https://pan.baidu.com/s/1Sc4nJ0a4FzFHBwoOpyGpUQ?pwd=6y7e

**Comment by yfyeung** [comment]
See: https://github.com/k2-fsa/icefall/pull/1500
